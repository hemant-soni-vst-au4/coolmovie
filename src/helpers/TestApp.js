import React from 'react';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { Router } from 'react-router-dom';
import Routes from '../routes';
import reducer from '../reducers/index';

const store = createStore(reducer, applyMiddleware(thunk));
const history = createBrowserHistory();
const TestApp = () => (
  <Router history={history}>
    <Provider store={store}>
      <Routes store={store} />
    </Provider>
  </Router>
);

export default TestApp;
