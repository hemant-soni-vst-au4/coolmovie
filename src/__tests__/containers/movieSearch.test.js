import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import TestApp from '../../helpers/TestApp';

describe('Render the page correctly', () => {
  test('Render correctly', () => {
    const { asFragment } = render(<TestApp />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Application name in place', () => {
    expect('MovieNchill').toMatchSnapshot();
  });

  test('Footer content in place', () => {
    expect('Created by Hemant Soni. Powered by ').toMatchSnapshot();
  });

  test('App header in place', () => {
    const { container } = render(<TestApp />);
    expect(container.innerHTML).toContain('<h3>MovieN<span>chill</span></h3>');
  });

  test('upcoming button in place', () => {
    const { container } = render(<TestApp />);
    expect(container.innerHTML).toContain('<button type="button" class="active">Upcoming</button>');
  });

  test('top rated button in place', () => {
    const { container } = render(<TestApp />);
    expect(container.innerHTML).toContain('<button type="button" class="unactive">Top Rated</button>');
  });

  test('Footer element image in place', () => {
    const { container } = render(<TestApp />);
    expect(container.innerHTML).toContain('<img style="height: 50px;" src="moviedblogo.svg" alt="moviedblogo">');
  });
});
