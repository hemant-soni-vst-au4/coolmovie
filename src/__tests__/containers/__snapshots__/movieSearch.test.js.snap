// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the page correctly Application name in place 1`] = `"MovieNchill"`;

exports[`Render the page correctly Footer content in place 1`] = `"Created by Hemant Soni. Powered by "`;

exports[`Render the page correctly Render correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <div
      class="search-elements"
    >
      <div
        style="display: flex; align-items: baseline;"
      >
        <img
          alt="movie'n'chill"
          src="popcorn.png"
          style="height: 50px; margin-left: 10px;"
        />
        <h3>
          MovieN
          <span>
            chill
          </span>
        </h3>
      </div>
      <input
        name="search"
        placeholder="Search for movie"
        value=""
      />
    </div>
    <div
      class="search-results"
    />
     
  </div>
  <div
    class="carousel-container"
    data-wow-duration="3s"
  >
    <div
      class="swiper-container swiper-container-initialized swiper-container-horizontal"
    >
      <div
        class="swiper-wrapper"
        style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
      />
      <div
        aria-label="Previous slide"
        class="swiper-button-prev"
        role="button"
        tabindex="0"
      />
      <div
        aria-label="Next slide"
        class="swiper-button-next"
        role="button"
        tabindex="0"
      />
      <span
        aria-atomic="true"
        aria-live="assertive"
        class="swiper-notification"
      />
    </div>
    <hr
      class="carousel-container__separator"
    />
  </div>
  <div
    class="container text-center"
  >
    <button
      class="active"
      type="button"
    >
      Upcoming
    </button>
    <button
      class="unactive"
      type="button"
    >
      Top Rated
    </button>
    <div
      class="movie-list mt-5"
    >
      <div
        class="container text-center"
      >
        <h2>
           UPCOMING 
        </h2>
        <div
          class="row"
          style="justify-content: center;"
        />
        <hr
          class="cards-separator"
        />
      </div>
       
    </div>
  </div>
  <div
    class="footer"
  >
    <p>
      Created by Hemant Soni. Powered by 
      <img
        alt="moviedblogo"
        src="moviedblogo.svg"
        style="height: 50px;"
      />
    </p>
  </div>
</DocumentFragment>
`;
